packages/playwright/src/AxeCoreWrapper/DefaultAxeCoreWrapper.cs

```diff
@@ -203,44 +208,48 @@ bool iframes
                 return partialResults;
             }
 
-            var frameContexts = await GetFrameContexts(frame, context);
+            var frameContexts = await GetFrameContexts(frame, context).ConfigureAwait(false);
+            var partialResultTasks = new List<Task<List<object?>>>();
             foreach (var fContext in frameContexts)
             {
                 try
                 {
                     object frameContext = JsonConvert.SerializeObject(fContext.Context);
                     string frameSelector = JsonConvert.SerializeObject(fContext.Selector);
-                    var frameHandle = await frame.EvaluateHandleAsync(EmbeddedResourceProvider.ReadEmbeddedFile("shadowSelect.js"), frameSelector);
+                    var frameHandle = await frame.EvaluateHandleAsync(await EmbeddedResourceProvider.ReadEmbeddedFileAsync("shadowSelect.js").ConfigureAwait(false), frameSelector).ConfigureAwait(false);
                     if (frameHandle != null)
                     {
                         var childFrameElement = frameHandle.AsElement();
                         if (childFrameElement == null)
                         {
-                            partialResults.Add(null);
+                            partialResults.Add(Task.FromResult<List<object?>>(new List<object?>(null)));
                             continue;
                         }
 
-                        var childFrame = await childFrameElement.ContentFrameAsync();
+                        var childFrame = await childFrameElement.ContentFrameAsync().ConfigureAwait(false);
                         if (childFrame == null)
                         {
-                            partialResults.Add(null);
+                            partialResults.Add(Task.FromResult<List<object?>>(new List<object?>(null)));
                             continue;
                         }
-
-                        partialResults.AddRange(await RunPartialRecursive(childFrame, options, frameContext, false, iframes));
+                        partialResultTasks.Add(RunPartialRecursive(childFrame, options, frameContext, false, iframes));
                     }
                     else
                     {
-                        Console.WriteLine("Frame Handle is null!!!!");
-                        partialResults.Add(null);
+                        partialResults.Add(Task.FromResult<List<object?>>(new List<object?>(null)));
                     }
                 }
                 catch (Exception)
                 {
-                    partialResults.Add(null);
+                    partialResults.Add(Task.FromResult<List<object?>>(new List<object?>(null)));
                 }
             }
 
+            foreach (var partialResultsList in await Task.WhenAll(partialResultTasks).ConfigureAwait(false))

```

Comment:
I don't think `Task.WhenAll` is quite right here; if an individual task fails, we should probably be applying the same "insert a null marker result for that one frame" behavior as the rest of the child frame functionality, but this would instead just throw an exception when `await`ing the `WhenAll` result. I think something closer to this would make more sense:

```cs
            foreach (var childPartialResultsTask in partialResultTasks)
            {
                List<object?> childPartialResults;
                try {
                    childPartialResults = await childPartialResultsTask.ConfigureAwait(false);
                } catch {
                    childPartialResults = new List<object?> { null };
                }   
                partialResults.AddRange(childPartialResults);
            }
```